### PR TITLE
Add gpg/libgcrypt version upgrade check in gpg test

### PR DESCRIPTION
**Pull request by jouyingbin**
Add record_info of version in gpg and version upgrade check for libgcrypt

**Libgcrypt needs to be upgraded to 1.9.x from SLE15 SP4**

- Related ticket: https://progress.opensuse.org/issues/104556
- Verification run: 
  (SLE 15 SP4) 
https://openqa.suse.de/tests/7939276#step/gpg/4 (gpg 2.2.27)
https://openqa.suse.de/tests/7939276#step/gpg/9 (libgcrypt 1.9.4)

  (SLE 15 SP3) 
https://openqa.suse.de/tests/7939277#step/gpg/4  (gpg 2.2.27)
https://openqa.suse.de/tests/7939277#step/gpg/9  (libgcrypt 1.8.2)

  (openSUSE TW) 
https://openqa.opensuse.org/tests/2121149#step/gpg/4  (gpg 2.3.4)
https://openqa.opensuse.org/tests/2121149#step/gpg/9 (libgcrypt 1.9.4)

### CODE REVIEW

**Review by rfan1 — APPROVED** [review]
LGTM

**Review by chcao — APPROVED** [review]
LGTM

File: tests/console/gpg.pm [hunk 1]
```diff
@@ -155,6 +156,19 @@ sub run {
     # Obtain GnuPG version
     my $gpg_version_output = script_output("gpg --version");
     my ($gpg_version) = $gpg_version_output =~ /gpg \(GnuPG\) (\d\.\d)/;
+    record_info('gpg version', "Version of Current gpg package: $gpg_version");
```
Comment (Amrysliu) [inline tests/console/gpg.pm:159]: According to the verify run https://openqa.opensuse.org/tests/2120886#step/gpg/2 , the gpg version is '2.3.4', but in the record message it is '2.3'. There is something wrong in the above regex. Please check that.

**Review by Amrysliu — COMMENTED** [review]
Others look good to me.

**Review by lilyeyes — APPROVED** [review]
lgtm.

File: tests/console/gpg.pm (hunk 1 above)
Comment (jouyingbin) [reply tests/console/gpg.pm:159]: Great comment, I think I miss 1 digit check, let me fix it, thanks @Amrysliu

Comment (jouyingbin) [reply tests/console/gpg.pm:159]: All VR is finished, resolved.

**Review by Amrysliu — APPROVED** [review]

**Pull request merged**